src/rgw/rgw_op.cc

```diff
@@ -4797,8 +4797,23 @@ void RGWPutObj::execute(optional_yield y)
     return;
   }
 
+  auto ret = rgw::bucketlogging::log_record(driver,
+      rgw::bucketlogging::LoggingType::Standard,
+      s->object.get(),
+      s,
+      (multipart ? "REST.PUT.PART" : canonical_name()),
```

Comment:
`RGWPutObj::canonical_name()` should probably handle this part itself. it can make the decision based on member variable `multipart_upload_id`